src/pip/_internal/utils/filesystem.py

```diff
@@ -17,14 +17,22 @@
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING, cast
 
 if MYPY_CHECK_RUNNING:
-    from typing import Any, BinaryIO, Iterator, List, Union
+    import abc
+    from typing import Any, BinaryIO, Iterator, List, Union, AnyStr
 
     class NamedTemporaryFileResult(BinaryIO):
         @property
         def file(self):
             # type: () -> BinaryIO
             pass
 
+        # Redefine the type annotations of BinaryIO.write to accept
+        # AnyStr as input instead of Union[bytes, bytearray]
+        @abc.abstractmethod
+        def write(self, s):
+            # type: (AnyStr) -> int
+            pass
```

Comment:
This doesn’t look right. Wouldn’t this accept `str` on Python 3? Maybe use `typing.ByteString` instead?

Reply:
Resolving this since casting at call-site is a better idea to be implemented, rather than changing types of `NamedTemporaryFileResult ` here.